pkgs/servers/search/sphinxsearch/default.nix

```diff
@@ -1,31 +1,44 @@
-{ stdenv, fetchurl, pkgconfig,
-  version ? "2.2.11",
-  mainSrc ? fetchurl {
-    url = "http://sphinxsearch.com/files/sphinx-${version}-release.tar.gz";
-    sha256 = "1aa1mh32y019j8s3sjzn4vwi0xn83dwgl685jnbgh51k16gh6qk6";
-  }
+{ stdenv, fetchurl, pkgconfig, expat, libmysqlclient,
+  enableXmlpipe2 ? false,
+  enableMysql ? true
 }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "sphinxsearch";
-  inherit version;
-  src = mainSrc;
+  version = "2.2.11";
+  src = fetchurl {
+    url = "http://sphinxsearch.com/files/sphinx-${version}-release.tar.gz";
+    sha256 = "1aa1mh32y019j8s3sjzn4vwi0xn83dwgl685jnbgh51k16gh6qk6";
+  };
+
+  enableParallelBuilding = true;
 
   configureFlags = [
     "--program-prefix=sphinxsearch-"
-    "--without-mysql"
     "--enable-id64"
+  ] ++ (if enableMysql then [
+    "--with-mysql=${libmysqlclient}/"
```

Comment:
Can this work with

```suggestion
    "--with-mysql"
```

Reply:
As far as my previous PR goes configure didn't find anything in that case. And I don't see why it should be different here.

Reply:
it'll actually find the lib without any parameter :) I changed it to just use `--without-mysql` if the feature's not enabled. The optionals reflect the default expected configuration in this way.

Reply:
That's really weird, I'm absolutely positive that it didn't work previously and I've tested it many times before adding that variable. I can't explain this and I'd better retest your expression in my environment to avoid any surprises.

Reply:
That's really weird, I'm absolutely positive that it didn't work previously and I've tested it many times before adding that variable. I can't explain this and I'd better retest your expression in my environment to avoid any surprises.

Reply:
Yup, that's likely it. That also explains why you needed the explicit dependencies on openssl and zlib, which are propagated by libmysqlclient. Fairly sure depending on the client lib instead of a whole database server is the way to go here, though ;)